Week3/homework/step2-4.js

```diff
@@ -1,7 +1,9 @@
 'use strict';
 
 function Dog() {
-  // add your code here
+  this.name = 'Bull';
+  this.color = 'Brown';
+  this.numLegs = 4;
 }
 
 const hound = new Dog();
```

Comment:
constructor functions used to be a way for creating objects in a more dynamic easy short way, yup all of these!

means if you have the **same** structure of data all time, as 
```javascript
{ name, age, favoritePlace, address1, address2, phoneNumber, mobileNumber, ...etc }
```
and they are with a different values each, imagine your self defining each of them in it's own line, as you are doing here on lines 4,5,6 , it's very short to have for a one user or two, but think of a million!

Instead of that we have constructors as:
```javascript
function User(name, age, favoritePlace, address1, address2, phoneNumber, mobileNumber) {
  this.name = name;
  this.age = age;
  this.favoritePlace = favoritePlace;
  this.address1 = address1;
  // ...etc
}

// And. we can have them all defined in a one line,
// as for now, let's make it simple and hard code them as required in the homework
const me = new user('Jawhar', 23, 'Home', 'some where' /*, ...etc */);
console.log(me) // this will give us the previous object with the specified details I wrote up above.
```

Same applies to the `Dog` Object constructor.